cts.json

```diff
@@ -2239,14 +2240,16 @@
       "result": [{"a": "bc"}]
     },
     {
-      "name": "filter, match function, invalid first arg",
+      "name": "filter, match function, non-string first arg",
       "selector" : "$[?match(1, 'a.*')]",
-      "invalid_selector": true
+      "document" : [{"a": "bc"}],
+      "result": []
     },
     {
-      "name": "filter, match function, invalid second arg",
+      "name": "filter, match function, non-string second arg",
       "selector" : "$[?match(@.a, 1)]",
-      "invalid_selector": true
+      "document" : [{"a": "bc"}],
```

Comment:
Change to `[{"a": 1}]` maybe?

Reply:
Why?  The point is that it doesn't select the item because `1` is not a string.

Reply:
But the expression isn't governed by the underlying engine.  It's our own syntax, and we don't have type coercion.

Also for the type system as it currently is, `1` and `"1"` are two unrelated values within the same type: `ValueType`.

Reply:
I think it would be better to have a test for `1=="1"` to test for that specifically.

Reply:
The second case only applies when there is coercion, and the tests I added in #23 ensure that doesn't happen.  It's not necessary to check that here.

Reply:
Some general comments:

* Each test should aim to test one thing.

* We may be able to borrow some tests from the JSONPath comparison project, or at least get inspiration from there, particularly for tests that check that non-standard behaviour does not occur. (@cburgmer may care to comment.)